languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/MiddlewareArg.java

```diff
@@ -0,0 +1,37 @@
+package client;
+
+public final class MiddlewareArg {
+
+    private final ApiHttpRequest request;
+    private final ApiHttpResponse response;
+    private final Exception error;
+    private final Middleware next;
+
+    private MiddlewareArg(ApiHttpRequest request, ApiHttpResponse response, Exception error, Middleware next) {
+        this.request = request;
+        this.response = response;
+        this.error = error;
+        this.next = next;
+    }
+
+    public static MiddlewareArg from(ApiHttpRequest request, ApiHttpResponse response, Exception error, Middleware next) {
```

Comment:
As already mentioned above: I would suggest to add another `from`factory method that only accepts an ÀpiHttpRequest`. This way it's clear that you only need the request to start with.